extensions/microsoft-authentication/package.json

```diff
@@ -31,6 +31,36 @@
       {
         "label": "Microsoft",
         "id": "microsoft"
+      },
+      {
+        "label": "Azure Cloud",
+        "id": "azure-cloud"
+      }
+    ],
+    "configuration": [
+      {
+        "title": "Azure Cloud",
+        "properties": {
+          "azure-cloud.endpoint": {
+            "anyOf": [
+              {
+                "type": "string"
+              },
+              {
+                "type": "string",
+                "enum": [
+                  "https://login.chinacloudapi.cn/",
+                  "https://login.microsoftonline.us/"
+                ],
+                "enumDescriptions": [
+                  "Azure China",
+                  "Azure US Government"
+                ]
```

Comment:
Could the enum value be "Azure China" and "Azure US Government" So that we can display that nicely as part of the Auth Provider label? Then we have a map of the names to their url values.

Reply:
I'd prefer my suggestion in this case. The docs for this setting could say `a preset like "Azure China" or a url to a custom cloud`

Reply:
Ok, added it as such. It still uses the `anyOf` with a regular `"type": "string"`, so the dropdown doesn't appear in the settnigs UI, but #178850 should fix that. You can access the presets by doing Ctrl+Space in the JSON editor:

![image](https://user-images.githubusercontent.com/36966225/229917094-576eae97-bf7b-4408-8895-cafca3fca6ec.png)

Reply:
@bwateratmsft can you get rid of the null entry? I think you can probably do this by having the default be `""` maybe...

Reply:
I tried that, here's what shows:
![image](https://user-images.githubusercontent.com/36966225/230624614-ed7f134c-acd0-458a-9f23-1a75dd14f5c2.png)

I don't feel strongly one way or the other.

Reply:
I tried putting the `"default": ""` in three different places--at the property root, and in each of the `"anyOf"` objects, and the results were either the same as above, or would show both `null` and `""`. We definitely don't want both.

Reply:
Ok null is fine, I guess. Now we really need that Settings Editor issue